### PR TITLE
Refactor RuleTypeMapping

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fix alignment in the rule type mapping.
Remove needed whitespace.
Store the full rule ID `S1234` instead of just the number `1234`. Because all usages were just adding it back anyway.

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeMappingCS.cs [hunk 1]
```diff
@@ -0,0 +1,6928 @@
+﻿/*
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeMappingCS.cs:1]: I tried to do it as a rename in a separate commit, but git didn't persisted that here

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/CsRuleTypeMapping.cs [hunk 2]
```diff
@@ -1,7000 +0,0 @@
...
- * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- */
-
-namespace SonarAnalyzer.UnitTest.PackagingTests
-{
-    internal static class CsRuleTypeMapping
-    {
-        #region RuleTypesCs
-#pragma warning disable SA1005, SA1507 // SingleLineCommentsMustBeginWithSingleSpace, CodeMustNotContainMultipleBlankLinesInARow
-        // The empty lines below are so the rule number aligns nicely with the line number.
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/CsRuleTypeMapping.cs:27]: The empty lines are practically not needed, we don't modify this manually. And were out of sync anyway.

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs [hunk 3]
```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
...
                                   {
-                                      ExpectedType = expectedTypes.GetValueOrDefault(i.ToString()),
-                                      ActualType = resourceManager.GetString($"S{i}_Type"),
-                                      RuleId = i
+                                      ExpectedType = expectedTypes.GetValueOrDefault(x),
+                                      ActualType = resourceManager.GetString($"{x}_Type"),
+                                      RuleId = x
                                   })
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:54]: The `x` changes its type in the query. Consider renaming the x to something more expressive. Also `var items` is also too generic. `RuleId` should be the first property in the anonymous type because it is the key of the list. I think this is easier to read.

```cs
var resourcesMissmatches = (from i in Enumerable.Range(1, 10000)
                            let ruleId = $"S{i}"
                            let matching = new
                            {
                                RuleId = ruleId,
                                ExpectedType = expectedTypes.GetValueOrDefault(ruleId),
                                ActualType = resourceManager.GetString($"{ruleId}_Type")
                            }
                            where matching.ActualType != matching.ExpectedType
                            select matching).ToList();
```

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs [hunk 4]
```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
...
 
-            // IMPORTANT: If this test fails, you should add the types of the new rules
-            // in the dictionaries above. It is a manual task, sorry.
-            var newRules = items.Where(x => x.ExpectedType == null);
+            // If this test fails, you should add the types of the new rules in the dictionaries above. It is a manual task, sorry.
+            var newRules = items.Where(x => x.ExpectedType is null);
             newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:58]: I can't remember: Does FluentAssertions outputs the entries in the list if `BeEmpty` fails. If not, the message should name the RuleIds that need to be added.

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs [hunk 5]
```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
...
+                                      ActualType = resourceManager.GetString($"{x}_Type"),
+                                      RuleId = x
                                   })
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
 
-            // IMPORTANT: If this test fails, you should add the types of the new rules
-            // in the dictionaries above. It is a manual task, sorry.
-            var newRules = items.Where(x => x.ExpectedType == null);
+            // If this test fails, you should add the types of the new rules in the dictionaries above. It is a manual task, sorry.
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:56]: > It is a manual task, sorry.

Is this true? I think the script does this for us.

Comment (martin-strecker-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:56]: Why is this comment not part of the assertation message?

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs [hunk 6]
```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
...
+            var newRules = items.Where(x => x.ExpectedType is null);
             newRules.Should().BeEmpty($"you need to add the rules in {expectedTypesName}");
 
-            // IMPORTANT: Rules should not be deleted without careful consideration and prior
-            // deprecation. We need to notify the product teams (SQ, SC, SL) as well.
-            items.Should().NotContain(x => x.ActualType == null && !deletedRules.Contains($"S{x.RuleId}"), "YOU SHOULD NEVER DELETE RULES!");
+            // Rules should not be deleted without careful consideration and prior deprecation. We need to notify the product teams (SQ, SC, SL) as well.
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:60]: Consider putting this comment in the assertation message

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs [hunk 7]
```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
...
+            items.Should().NotContain(x => x.ActualType == null && !DeletedRules.Contains($"S{x.RuleId}"), "YOU SHOULD NEVER DELETE RULES!");
 
-            // IMPORTANT: If this test fails, you should update the types of the changed rules
-            // in the dictionaries above. Also add a GitHub issue specifying the change of type
-            // and update peach and next.
+            // If this test fails, you should update the types of the changed rules in the dictionaries above. Also add a GitHub issue specifying the change of type and update peach and next.
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:63]: This too

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
It would have been nice to have a PR description of what this is about. It took me a while to figure out, that the mappings were not just file renames.

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs (hunk 4 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:58]: Yes, it does.

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs (hunk 5 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:56]: It seems redundant in this assertion. I've removed it.

File: analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs:54]: The `x` is easy to follow. I've reordered the items.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM
